src/structured_data.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use rustc_serialize::{Encodable, Encoder, Decoder};
+use cbor;
+use error::RoutingError;
+use NameType;
+use sodiumoxide::crypto;
+use std::str;
+
+/// StructuredData
+/// These types may be stored unsigned with previous and current owner keys
+/// set to the same keys. Updates though require a signature to validate
+#[derive(Clone, RustcDecodable, RustcEncodable)]
+pub struct StructuredData {
```

Comment:
4. Yes, I don't like it in the rest of the code neither but especially here.

The main reason is that we'll now have two instances of the field, namely `None` and `Some(empty_vec)` to mean exactly the same thing. So instead of asking `if !owners.is_empty() ...` you'll be asking `if owners.is_some() && !owners.is_empty() ...`. And the reason I don't like it 'especially' in this case is that it adds more data to the message.